lib/SIL/IR/SILFunctionType.cpp

```diff
@@ -3423,6 +3423,26 @@ class CFunctionTypeConventions : public Conventions {
                            const clang::FunctionType *type)
     : Conventions(kind), FnType(type) {}
 
+  // Helper function to return ResultConvention if the decl has either
+  // swift_attr("returns_retained") or ("returns_unretained") attribute.
+  std::optional<ResultConvention>
+  getResultConventionWithAttributes(const TypeLowering &tl,
+                                    const clang::FunctionDecl *decl) const {
```

Comment:
Two minor comments:
1. "Helper function to return ResultConvention" doesn't add much value, since this information is already in the function's signature. Perhaps we could rephrase this comment as something like "Determines owned/unowned convention of the returned value based on returns_retained/returns_unretained attribute"?
2. Would the meaning of this function be more clear if it was renamed it to `getForeignReferenceTypeResultConvention`?

Reply:
1.) -> Agreed.
2.) `getForeignReferenceTypeResultConvention` or  `getForeignReferenceTypeResultConventionWithAttributes` ? Because this function doesn't cover the logic when FRT are handled without attributes. The default behavior (without attributes) are not changed in this patch.